src/lib/pbkdf/scrypt/scrypt.cpp

```diff
@@ -83,18 +83,20 @@ std::unique_ptr<PasswordHash> Scrypt_Family::tune(size_t output_length,
 
    uint64_t est_nsec = measured_time;
 
-   // First move increase r by 8x if possible
+   // In below code we invoke scrypt_memory_usage with p == 0 as p contributes
+   // (very slightly) to memory consumption, but N is the driving factor.
+   // Including p leads to using an N half as large as what the user would expect.
 
-   if(max_memory_usage == 0 || scrypt_memory_usage(N, r * 8, p) <= max_memory_usage) {
+   // First move increase r by 8x if possible
```

Comment:
```suggestion
   // First increase r by 8x if possible
```
... I guess.

Reply:
Fixed that and also a now-incorrect comment regarding the removed addition of 512 to the memory bound